src/react/useActiveFile.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+import { session } from "src/api";
+import useReplit from "./useReplit";
+
+export default function useActiveFile() {
+  const [file, setFile] = useState<string | null>(null);
+
+  const { status } = useReplit();
+
+  useEffect(() => {
+    if (status === "ready") {
+      (async () => {
+        setFile(await session.getActiveFile());
+      })();
+
+      return session.onActiveFileChange(async (f) => setFile(f));
+    } else {
+      return () => {};
+    }
+  }, [status]);
```

Comment:
- the hook might be unmounted here before the async function returns, and we should not call setFile in that case.
- early return makes things more readable (faris, if you're reading this, i hope you are proud)

```suggestion
  useEffect(() => {
    if (status !== "ready") {
      return
    }
  
    let cancelled = false;
    (async () => {
      const activeFile = await session.getActiveFile();
      
      if (cancelled) {
        return;
      }
      
      setFile(activeFile);
    })();

    const dispose = session.onActiveFileChange((f) => setFile(f));
    
    return () => {
      cancelled = true;
      dispose();
    }
  }, [status]);
```